kuksa_databroker/databroker/src/broker.rs

```diff
@@ -1333,11 +1332,16 @@ impl<'a, 'b> AuthorizedAccess<'a, 'b> {
         let valid_entries = {
             let mut valid_entries = HashMap::new();
             for (path, fields) in entries {
-                match self.get_id_by_path(path.as_ref()).await {
-                    Some(id) => {
-                        valid_entries.insert(id, fields);
-                    }
-                    None => return Err(SubscriptionError::NotFound),
+                let regex_exp = glob::to_regex(&path);
+                if let Ok(regex) = regex_exp {
+                    self.for_each_entry(|entry| {
+                        let entry_fields = fields.clone();
+                        let entry_path = &entry.metadata().path;
+                        if regex.is_match(entry_path) {
+                            valid_entries.insert(entry.metadata().id, entry_fields);
+                        }
+                    })
+                    .await;
```

Comment:
Code updated with your approach

Reply:
Looks better.

It still doesn't handle the case when trying to subscribe to a non existing path.
Since everything is treated as a wildcard pattern, I guess it would have to be solved by checking if "wildcard pattern matches nothing" -> error "not found".